subsys/bluetooth/controller/ll_sw/ull_scan.c

```diff
@@ -510,6 +727,170 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t remainder, uint16_t laz
 	DEBUG_RADIO_PREPARE_O(1);
 }
 
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+static uint8_t is_scan_update(uint8_t handle, uint16_t duration,
+			      uint16_t period, struct ll_scan_set **scan,
+			      struct node_rx_pdu **node_rx_scan_term)
+{
+	*scan = ull_scan_set_get(handle);
+	*node_rx_scan_term = (void *)(*scan)->node_rx_scan_term;
+	return duration && period && (*scan)->lll.duration_reload &&
+	       (*scan)->duration_lazy;
```

Comment:
I did not find in the spec that old or new duration parameters need to be non-zero for a scan update. But maybe I am overlooking something?

Reply:
BT Spec v5.2, Vol.4, Part E, Section 7.8.65

> When the Duration and Period parameters are non-zero, the Controller shall scan for the duration of the Duration parameter within a scan period specified by the Period parameter.

and 

> If the HCI_LE_Set_Extended_Scan_Enable command with Enable set to 0x01 is sent while scanning is already enabled, the timers used for duration and period are reset to the new parameter values and a new scan period is started.

For the update to be allowed both duration and period need to be non-zero. As Period parameter specifies 'scan period'.

Reply:
Do I understand correctly that you interpret this such that the "new scan period" mentioned in the spec implies that there must be an "old scan period"? The test specification does not seem to test this behavior, btw.

Reply:
> you interpret this such that the "new scan period" mentioned in the spec implies that there must be an "old scan period"

Yes.

> The test specification does not seem to test this behavior, btw.

I did not get, there is a test that wants to switch from a scanning with non-zero Duration and zero Period to both non-zero and vice versa? Could you point me to the test cases that I can refer to, to make changes accordingly.

Reply:
Looks like I did not express myself clearly... I could not find any test case that uses a non-zero period, and only one that uses a non-zero duration. There does not seem to be any test case that updates these parameters. We therefore cannot use the test spec to infer the intended behavior. I do not find the wording in the BT Spec crystal clear, but I think your interpretation is valid.

Reply:
If the implementation needs to allow updates to scanning irrespective of old duration and period, its is not difficult to fix it based on a requirement/request from users. Additonal implementation is required to release buffers allocated for scan terminate event, if period is updated from zero to non-zero value.